src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceSupervisor.cs

```diff
@@ -269,6 +275,35 @@ private void ListenToMaintenanceWorker()
                 }
             }
 
+            private void UpdateNodeReportIfNeeded(ClusterNodeStatusReport nodeReport, List<DatabaseStatusReport> unchangedReports)
+            {
+                foreach (var dbReport in nodeReport.Report)
+                {
+                    if (dbReport.Value.Status == DatabaseStatus.NoChange)
+                    {
+                        unchangedReports.Add(dbReport.Value);
+                    }
+                }
+
+                if (unchangedReports.Count == 0)
+                    return;
+
+                foreach (var dbReport in unchangedReports)
+                {
+                    var dbName = dbReport.Name;
+                    if(_lastSuccessfulReceivedReport.TryGetValue(name, out var previous) == false)
```

Comment:
@karmeli87 Need to see if the caller code can handle the case where we get the unchanged directly.